File: bindings/csharp/src/ProvenGit.cs
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// Git protocol types for proven-servers.

namespace Proven;

/// <summary>Command matching the Idris2 ABI tags (0-2).</summary>
public enum Command : byte
{
    UploadPack = 0,
    ReceivePack = 1,
    UploadArchive = 2
}

/// <summary>PacketType matching the Idris2 ABI tags (0-7).</summary>
public enum PacketType : byte
{
    Flush = 0,
    Delimiter = 1,
    ResponseEnd = 2,
    Data = 3,
    PktError = 4,
    SidebandData = 5,
    SidebandProgress = 6,
    SidebandError = 7
}

/// <summary>RefType matching the Idris2 ABI tags (0-4).</summary>
public enum RefType : byte
{
    Branch = 0,
    Tag = 1,
    Head = 2,
    Remote = 3,
    GitNote = 4
}

/// <summary>Capability matching the Idris2 ABI tags (0-8).</summary>
public enum Capability : byte
{
    MultiAck = 0,
    ThinPack = 1,
    SideBand64k = 2,
    OfsDelta = 3,
    Shallow = 4,
    DeepenSince = 5,
    DeepenNot = 6,
    FilterSpec = 7,
    ObjectFormat = 8
}

/// <summary>HookResult matching the Idris2 ABI tags (0-1).</summary>
public enum HookResult : byte
{
    Accept = 0,
    Reject = 1
}

/// <summary>ServerState matching the Idris2 ABI tags (0-4).</summary>
public enum ServerState : byte
{
    Idle = 0,
    Discovery = 1,
    Negotiating = 2,
    Transfer = 3,
    Shutdown = 4
}

File: bindings/php/src/ProvenGit.php
<?php
// SPDX-License-Identifier: MPL-2.0
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// Git protocol types for proven-servers.

declare(strict_types=1);

namespace ProvenServers;

/** Command matching the Idris2 ABI tags. */
enum Command: int
{
    case UploadPack = 0;
    case ReceivePack = 1;
    case UploadArchive = 2;
}

/** PacketType matching the Idris2 ABI tags. */
enum PacketType: int
{
    case Flush = 0;
    case Delimiter = 1;
    case ResponseEnd = 2;
    case Data = 3;
    case PktError = 4;
    case SidebandData = 5;
    case SidebandProgress = 6;
    case SidebandError = 7;
}

/** RefType matching the Idris2 ABI tags. */
enum RefType: int
{
    case Branch = 0;
    case Tag = 1;
    case Head = 2;
    case Remote = 3;
    case GitNote = 4;
}

/** Capability matching the Idris2 ABI tags. */
enum Capability: int
{
    case MultiAck = 0;
    case ThinPack = 1;
    case SideBand64k = 2;
    case OfsDelta = 3;
    case Shallow = 4;
    case DeepenSince = 5;
    case DeepenNot = 6;
    case FilterSpec = 7;
    case ObjectFormat = 8;
}

/** HookResult matching the Idris2 ABI tags. */
enum HookResult: int
{
    case Accept = 0;
    case Reject = 1;
}

/** ServerState matching the Idris2 ABI tags. */
enum ServerState: int
{
    case Idle = 0;
    case Discovery = 1;
    case Negotiating = 2;
    case Transfer = 3;
    case Shutdown = 4;
}
